js/pin.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+(() => {
+  const mapPinsNode = document.querySelector(`.map__pins`);
+  const mapNode = document.querySelector(`.map`);
+  const mapPinNode = document.querySelector(`#pin`)
+    .content
+    .querySelector(`.map__pin`);
```

Comment:
Единообразия ради
```suggestion
const mapPinNode = document.querySelector(`#pin`).content.querySelector(`.map__pin`);
```

также стоит указать, что это шаблон, а не рядовой ДОМ элемент